src/main/java/sa/assertj/iterables/AllSatisfy.java

```diff
@@ -0,0 +1,35 @@
+package sa.assertj.iterables;
+
+import sa.assertj.Experiment;
+
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AllSatisfy extends Experiment {
+    static Experiment.DataProvider provider = (size, numOfSamples) -> {
+        Object[][] result = new Object[numOfSamples][];
+        for (int i = 0; i < numOfSamples; i++) {
+            ArrayList<Integer> list = new ArrayList<>();
+            Random rand = new Random();
+            int element = rand.nextInt();
+            for (int j = 0; j < size; j++) {
+                list.add(element);
+            }
+
+            Consumer<Integer> consumer = s -> s.equals(element);
```

Comment:
According to the example in the AssertJ documentation, this is the use case:

```java
assertThat(myIcelanderFriends).allSatisfy(person -> {
                                     assertThat(person.getCountry()).isEqualTo("Iceland");
                                     assertThat(person.getPhoneCountryCode()).isEqualTo("+354");
                                   });
```

Maybe we could adapt it to something like that? Cause the point is that this takes a void method unlike `AllMatch` so e.g.

```java
i -> {
  assertThat(i).isGreaterThanOrEqualTo(Integer.MIN_VALUE);
  assertThat(i).isLessThanOrEqualTo(Integer.MAX_VALUE);
}
```
or simply just:
```java
i -> assertThat(i).isEqualTo(element);
```